packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -112,17 +115,38 @@ async function invokeUserFunction<A extends { ResponseURL: '...' }>(functionArnE
 
   log('user function response:', resp, typeof(resp));
 
+  // parse function name from arn
+  // arn:${Partition}:lambda:${Region}:${Account}:function:${FunctionName}
+  const arn = functionArn.split(':');
+  const functionName = arn[arn.length - 1];
```

Comment:
Is there a circumstance where this lambda might have a version added to the end of it or are our custom resources using unqualified ARNs?

Reply:
The user function is under the user's control, so they can create versions as they please. However, `getFunction()` will happily accept a name with a `:version-string` at the end as well, so we're safe in either case.